File: api/internal/domain/lead_conversion_mapping.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type LeadConversionTarget string

const (
	LeadConversionTargetContact LeadConversionTarget = "contact"
	LeadConversionTargetAccount LeadConversionTarget = "account"
	LeadConversionTargetDeal    LeadConversionTarget = "deal"
)

func (t LeadConversionTarget) IsValid() bool {
	return t == LeadConversionTargetContact || t == LeadConversionTargetAccount || t == LeadConversionTargetDeal
}

type LeadConversionMapping struct {
	ID           uuid.UUID            `json:"id"`
	OrgID        uuid.UUID            `json:"org_id"`
	LeadField    string               `json:"lead_field"`
	TargetEntity LeadConversionTarget `json:"target_entity"`
	TargetField  string               `json:"target_field"`
	IsActive     bool                 `json:"is_active"`
	CreatedAt    time.Time            `json:"created_at"`
	UpdatedAt    time.Time            `json:"updated_at"`
}

File: web/src/api/leadConversionMapping.ts
import apiClient from './client'

export type LeadConversionTarget = 'contact' | 'account' | 'deal'

export interface LeadConversionMapping {
  id?: string
  org_id?: string
  lead_field: string
  target_entity: LeadConversionTarget
  target_field: string
  is_active: boolean
  created_at?: string
  updated_at?: string
}

export const leadConversionMappingApi = {
  list: async (): Promise<LeadConversionMapping[]> => {
    const { data } = await apiClient.get('/settings/lead-conversion-mapping')
    return data.data ?? []
  },

  replace: async (mappings: LeadConversionMapping[]): Promise<LeadConversionMapping[]> => {
    const { data } = await apiClient.put('/settings/lead-conversion-mapping', { mappings })
    return data.data ?? []
  },
}
